interface/network_interface.cpp

```diff
@@ -0,0 +1,169 @@
+#include <cstring>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <net/if.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#include <sys/ioctl.h>
+
+#include "network_interface.h"
+#include "ilogger.h"
+#include "interface_addresses.h"
+
+NetworkInterface::NetworkInterface(const std::string& name, std::shared_ptr<ILogger> logger) :
+        m_logger(logger),
+        m_name(name)
+{
+    m_fd = socket(AF_INET, SOCK_DGRAM, 0);
+}
+
+NetworkInterface::~NetworkInterface()
+{
+    if (m_fd >= 0)
+    {
+        close(m_fd);
+    }
+}
+
+bool NetworkInterface::AddIpAddress(const std::string& ipAddress, const std::string& netmask) const
+{
+    if (m_fd < 0)
+    {
```

Comment:
this check should probably be moved to the constructor, as that is the place that creates m_fd. Also, since you use this multiple times in add ip address and remove ip address instead of once